### ISSUE TITLE
Tutorial 4 : AssertionError during the trainning 

### ISSUE THREAD

**Issue by WendkuuniArzouma** [bentrevett/pytorch-seq2seq#173]
Hello, when I run the code source with 100 epochs, after  the 53th I get **AssertionError:** without any message.
Below I attach some screenshots. I don't know how tocorrect it.

![bug4](https://user-images.githubusercontent.com/63440937/134076896-ffbbb04b-1998-40dc-b0b4-965cda435359.png)
![bug](https://user-images.githubusercontent.com/63440937/134076900-71345e27-f6b6-4fd3-95be-7fe98a0a11db.png)
![bug1](https://user-images.githubusercontent.com/63440937/134076906-f7274f09-7a43-43db-baa4-20431ad874c3.png)
![bug2](https://user-images.githubusercontent.com/63440937/134076908-1ec21cc6-0a93-416c-81a6-d92773cede69.png)
![bug3](https://user-images.githubusercontent.com/63440937/134076912-bf84fdb4-fe37-4ad4-8de2-e1aad6ffcefa.png)

**Comment by bentrevett** [comment]
I've never encountered this myself, but I believe it's because the gradients are either becoming NaN or infinity. 

Some potential fixes are lowering the learning rate, or using a better initialization strategy. I'll look into this to see if I can repoduce.

**Comment by WendkuuniArzouma** [comment]
> I've never encountered this myself, but I believe it's because the gradients are either becoming NaN or infinity.
> 
> Some potential fixes are lowering the learning rate, or using a better initialization strategy. I'll look into this to see if I can repoduce.

Okay, I understand a little better.
But concerning the learning rate, I looked for it but without success. I am a beginner, but usually I think it takes the name of variable **lr**, or **learning_rate**. But here I confess I'm lost.

**Comment by harshraj22** [comment]
@WendkuuniArzouma , cell num 16
```python
optimizer = optim.Adam(model.parameters())
```
Adam has a default learning rate parameter. If you want to change that, manually specify the learning rate. eg
```python
optimizer = optim.Adam(model.parameters(), lr=0.001)
```

**Comment by WendkuuniArzouma** [comment]
@harshraj22  many thanks for your help!
I understand now!

**Comment by WendkuuniArzouma** [comment]
Hello!!After modifying the lr as you suggested it works very well. Thanks again for your help.
Before closing the issue, I would like to ask one last question.
How to save the model so that it can be used in another machine directly for inference?

**Comment by harshraj22** [comment]
Following the [official documentation](https://pytorch.org/tutorials/beginner/saving_loading_models.html),
to save the model:

```python
torch.save(model.state_dict(), PATH)
```

```python
model = TheModelClass(*args, **kwargs)
model.load_state_dict(torch.load(PATH))
model.eval()
```

**Comment by WendkuuniArzouma** [comment]
@harshraj22  many thanks for your help!
I will follow the official documentation.
Thank you!!!
